### DISCUSSION TITLE
What error to return when SIM Swap info is technically not available for a particular phone number

### DISCUSSION THREAD

**Discussion by HuubAppelboom** [camaraproject/SimSwap#202] [General]
We have some very rare cases (less than 1 in 10000) where we don't have a SIM Activation time stamp in our platform. 
The mobile phone number is however valid, and belongs to KPN.

Currently we are returning a null value in stead of a time stamp in the reply body.

It may be better however to return an error code instead 
(also to avoid confusion also when there is a limitation becuase of privacy reasons).

What is the error code to return in this case?  

Should we use a 422 error with code NOT_SUPPORTED and message Service not supported for this phoneNumber ? 

Or would it be better to provide a 204 error code (No Content) ??

Please advice what is best.

**Comment by bigludo7** [comment]
I had the case in one Orange implem for very specific number where we were not able to get any date (and no able to provide the check). For this case we use the 422 with NOT_SUPPORTED code.

**Comment by maxl2287** [comment]
> I had the case in one Orange implem for very specific number where we were not able to get any date (and no able to provide the check). For this case we use the 422 with NOT_SUPPORTED code.

@bigludo7 or we use the CAMARA alligned Error code for 422 `SERVICE_NOT_APPLICABLE`

**Comment by bigludo7** [comment]
> > I had the case in one Orange implem for very specific number where we were not able to get any date (and no able to provide the check). For this case we use the 422 with NOT_SUPPORTED code.
> 
> [@bigludo7](https://github.com/bigludo7) or we use the CAMARA alligned Error code for 422 `SERVICE_NOT_APPLICABLE`

@maxl2287 Yes you're right.... and this error is specified in our yaml.

**Comment by bigludo7** [comment]
As not updated transferred to discussion
